### PR TITLE
Error handling from a single service

### PR DESCRIPTION
**Pull request by brick9527**
Fix problem:  If the node is disconnected, or the client does not have access to the internet, printing a warning instead of throwing an error

### CODE REVIEW

File: src/services/nodeError.service.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        const regexp = /^(Error: )?connect ECONNREFUSED/g
+        if (err.message.match(regexp)) {
+            const nodeIP = err.message.replace(regexp, '').trim()
+            console.log('\nError: Can\'t reach the node: ' + nodeIP)
+            process.exit()
+        }
+        err = err.message ? JSON.parse(err.message) : err
+        console.log(chalk.red('Error'), err.body && err.body.message ? err.body.message : err)
+        if (callback) {
+            return callback(err)
```
Comment (dgarcia360) [inline src/services/nodeError.service.ts]: Do not use callbacks. Also, there aren't any async calls inside this service to have to wait until this completes.

File: src/services/nodeError.service.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+     * Handle connect ECONNREFUSED error.
+     * @param {any} err - err data.
+     * @param {function} callback  - callback for command to complete this error handler.
+     */
+    public static connectErrorHandler(err: any, callback?: Function) {
+        const regexp = /^(Error: )?connect ECONNREFUSED/g
+        if (err.message.match(regexp)) {
+            const nodeIP = err.message.replace(regexp, '').trim()
+            console.log('\nError: Can\'t reach the node: ' + nodeIP)
+            process.exit()
```
Comment (dgarcia360) [inline src/services/nodeError.service.ts]: Do not exit the app.

File: src/services/nodeError.service.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+import chalk from 'chalk'
+
+export class NodeErrorService {
```
Comment (dgarcia360) [inline src/services/nodeError.service.ts]: Add unit test.

File: src/services/nodeError.service.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    /**
+     * Handle connect ECONNREFUSED error.
+     * @param {any} err - err data.
+     * @param {function} callback  - callback for command to complete this error handler.
```
Comment (dgarcia360) [inline src/services/nodeError.service.ts]: Update docs

**Comment by brick9527** [comment]
> I like the idea of handling the node error in a separate class. However, it's not currently solving the error reported for all the cases. For example, try to run `nem2-cli account info` without having an internet connection, and you will get the exception.

I realized this bug. I found it will throw error when using `forkJoin`, other operation is ok. I didn't know why cause this.

**Comment by decentraliser** [comment]
> > I like the idea of handling the node error in a separate class. However, it's not currently solving the error reported for all the cases. For example, try to run `nem2-cli account info` without having an internet connection, and you will get the exception.
> 
> I realized this bug. I found it will throw error when using `forkJoin`, other operation is ok. I didn't know why cause this.

I think this problem that comes from the SDK, the error is not properly handled in http.ts

**Comment by dgarcia360** [comment]
Added a commit on top making it easier to follow how errors are handled.

**Comment by dgarcia360** [comment]
Note: At the moment we are not fixing the ECONNREFUSED; this might be fixed with the next NEM2-SDK release.

**Pull request merged**